### PR TITLE
Use rancher catalog

### PR DESCRIPTION
**Pull request by adamgraves-choices**

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
@adamgraves-choices - I realise it's still WIP but was interested in it and it looks good.

I had a play wth the rancher API last week and came up with the following snippets which might simplify the script.

_Get the latest version id:_

```
RANCHER_TEMPLATE_VERSION=$( \
  curl -su "${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY}" "${RANCHER_SERVER}/v1-catalog/templates/${CATALOG_ID}:${RANCHER_TEMPLATE_NAME}" | \
  jq --raw-output '.defaultTemplateVersionId'
)
```

**Note**: I've assumed that default and latest are the same thing. Not sure if there is a scenario where default might not be the latest. To get a feel for the data being returned just do the above but change the jq filter to `jq '.'`

_Populate the answers file:_

```
curl -su "${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY}" "${RANCHER_SERVER}/v1-catalog/templates/${RANCHER_TEMPLATE_VERSION} | jq --raw-output '.files["rancher-compose.yml"]'  | \
python -c 'import sys, yaml, json; json.dump(yaml.load(sys.stdin), sys.stdout)' | \
jq --raw-output '.catalog.questions[] | @text "\(.variable)=\(.default)"' > answers.txt
```

File: rancher_pr_deploy.sh [hunk 1]
```diff
@@ -55,14 +55,18 @@ rancher() {
...
+
 if [ "$TRAVIS" == true ] && [ "$TRAVIS_PULL_REQUEST" != false ] ; then
 
   fold_start "Generate_answers.txt"
   # POPULATES ANSWERS FILE WITH THE DEFAULTS FROM THE RANCHER-COMPOSE FILE
   RANCHER_CATALOG_ID=$( curl -su "${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY}" "${RANCHER_SERVER}/v1-catalog/templates/${RANCHER_CATALOG_NAME}:${RANCHER_TEMPLATE_NAME}" | jq --raw-output '.defaultTemplateVersionId' )
   curl -su "${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY}" "${RANCHER_SERVER}/v1-catalog/templates/${RANCHER_CATALOG_ID}" | \
   jq --raw-output '.files["rancher-compose.yml"]' | \
-    python -c 'import sys, yaml, json; json.dump(yaml.load(sys.stdin), sys.stdout)' | \
+    yml2json | \
```
Comment (neilbmclaughlin) [inline rancher_pr_deploy.sh]: This means we no longer need the `pip install` line in the .travis.yml file

Comment (neilbmclaughlin) [reply rancher_pr_deploy.sh]: Also, not sure how you weigh up the benefits of running a single command (python/rancher/etc) in a docker container vs installing it in Travis. I guess (as you mentioned in the catchup) it makes the script more self contained and able to be run in environments other than Travis?

Comment (neilbmclaughlin) [reply rancher_pr_deploy.sh]: I which case should we be doing the same for docker-compose?

**Comment by neilbmclaughlin** [comment]
Running the command through shellcheck gave several warnings. Also, been bugging me the deployment to rancher didn't previously break the build. I've created a PR which resolves these issues here #26. Feel free to use it if you think it's useful.

File: rancher_pr_deploy.sh (hunk 1 above)
Comment (neilbmclaughlin) [reply rancher_pr_deploy.sh]: Finally, yml2json is broken. The -a flag needs STDIN/STDOUT and is redundent I think. Played around trying to get it to work for a bit but with no luck (the python yaml package needs installing in the container).

Comment (adamgraves-choices) [reply rancher_pr_deploy.sh]: We don't need to convert the YAML to json anymore anyway, so it's a moot point.

**Comment by adamgraves-choices** [comment]
Cheers @neilbmclaughlin, merged. Removed the `yaml2json` function as we can get the questions directly from the catalog API.

I've removed the WIP label, so when you're happy it's good to go.

**Review by neilbmclaughlin — APPROVED** [review]

**Pull request merged**